### PR TITLE
fix broken isFirefox on pages/*

### PR DESCRIPTION
**Pull request by gdh1995**
The commit 9005144cd02fab59577c24f50aae279f4160ce06 which modified #3460 may not work, because `Utils.isFirefox` was always `false` unless in Vimium's background / content_scripts page. So this PR passes the result of `isFirefox` from content scripts to `pages/*.html`, including the HUD.

### CODE REVIEW

**Comment by philc** [comment]
Thanks for catching! We can drop the trailing `if` statement on my commit if that makes this easier.

However, I'm confused by the complexity of the `isFirefox` detection mechanism and this PR. Why can't we just use the [userAgent](https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent), so it doesn't require information passing from background pages?

**Comment by gdh1995** [comment]
A web developer may override useragent through The Developer Tools, so it's not robust enough.


This PR is depended on by another of my PRs (the "ime in findmode")



---Original---

File: content_scripts/link_hints.coffee [hunk 1]
```diff
@@ -372,7 +372,7 @@ class LinkHintsMode
             Utils.nextTick -> focusThisFrame highlight: true
           else if localHintDescriptor.reason == "Scroll."
             # Tell the scroller that this is the activated element.
-            handlerStack.bubbleEvent "DOMActivate", target: clickEl
+            handlerStack.bubbleEvent (if Utils.isFirefox() then "click" else "DOMActivate"), target: clickEl
```
Comment (philc) [inline content_scripts/link_hints.coffee]: Where did this change come from? Seems unrelated to this PR.

File: content_scripts/ui_component.coffee [hunk 2]
```diff
@@ -72,6 +72,8 @@ class UIComponent
                 when "setIframeFrameId" then @iframeFrameId = event.data.iframeFrameId
                 when "hide" then @hide()
                 else @handleMessage event
+      if Utils.isFirefox()
```
Comment (philc) [inline content_scripts/ui_component.coffee]: Add a newline before this line, to separate it from the very long block before it.

File: content_scripts/link_hints.coffee (hunk 1 above)
Comment (gdh1995) [reply content_scripts/link_hints.coffee]: Unrelated. This is just to fix another issue that on Firefox `LinkHints` can not notify `Scroller` what to scroll.
